sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracer.java

```diff
@@ -189,14 +193,50 @@ public Context getSharedSpanBuilder(String spanName, Context context) {
         return context.addData(SPAN_BUILDER_KEY, getSpanBuilder(spanName, context));
     }
 
+    @Override
+    public void addEvent(String eventName, Map<String, Object> attributes, Instant timestamp) {
+        Span currentSpan = Span.current();
+        if (currentSpan == null) {
+            logger.info("Failed to find a starting span to associate an event with.");
+            return;
+        }
+
+        currentSpan.addEvent(
+            eventName,
+            attributes == null ? Attributes.empty() : convertToAttributes(attributes),
+            timestamp
+        );
+    }
+
+    /**
+     * Maps span/event properties to OpenTelemetry attributes.
+     *
+     * @param attributes the attributes provided by the client SDK's.
+     * @return the OpenTelemetry typed {@Link Attributes}.
+     */
+    private Attributes convertToAttributes(Map<String, Object> attributes) {
+        AttributesBuilder attributesBuilder = Attributes.builder();
+        attributes.forEach((key, value) -> {
+            if (value instanceof Boolean) {
```

Comment:
Or add handling to deal with unboxing of a null boxed primitive

Reply:
`instanceOf` should work for primitives as well.
As for null, it would just skip since it wouldn't be an instance of anything.
Added a test case for this https://github.com/Azure/azure-sdk-for-java/pull/18922/files#diff-07afbdde98ab2049a22d0ad5a6e1caf0ce9c4b34f939ec521a7d86f208765d91R477-R483